Mathlib/Algebra/Group/UniqueProds.lean

```diff
@@ -203,17 +215,138 @@ instance {M} [Mul M] [UniqueProds M] : UniqueSums (Additive M) where
 
 end Additive
 
+#noalign covariants.to_unique_prods
+#noalign covariants.to_unique_sums
+
+namespace UniqueProds
+
+variable {G H : Type*} [Mul G] [Mul H]
```

Comment:
Moving up these variables facilitates replacing `A` by `G`, avoiding confusion with the two Finsets A and B.